File: mesmaeker-java-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingThrowingClause.java
package fr.xgouchet.mesmaeker.mockito;

import java.util.function.Consumer;

/**
 * @author Xavier F. Gouchet
 */
public interface OngoingThrowingClause<T> extends OngoingReverseClause<T> {


    OngoingThrowingClause<T> when(Consumer<T> consumer);
}

File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/OngoingThrowingClause.kt
package fr.xgouchet.mesmaeker.mockito

/**
 * @author Xavier F. Gouchet
 */
interface OngoingThrowingClause<T> : OngoingReverseClause<T> {


    fun `when`(block: (T) -> Unit): OngoingThrowingClause<T>

    fun whenever(block: (T) -> Unit): OngoingThrowingClause<T>
}
